test/c2h/generators.cu

```diff
@@ -380,5 +380,65 @@ VEC_SPECIALIZATION(double, 2);
 VEC_SPECIALIZATION(uchar, 3);
 
 VEC_SPECIALIZATION(ulonglong, 4);
+
+
+template <int VecItem>
+struct set_vec_element_field_t;
+
+template <>
+struct set_vec_element_field_t<0>{
+template <typename T, typename C>
+static __host__ __device__ __forceinline__ void set(T& in_out, C c)
+{in_out.x = c;}
+};
+
+#define SET_VEC_ELEMENT_SPEC(VEC_ITEM, VEC_FIELD)                               \
+  template <>                                                                   \
+  struct set_vec_element_field_t<VEC_ITEM>{                                     \
+  template <typename T, typename C>                                             \
+  static __host__ __device__ __forceinline__ void set(T& in_out, C c)           \
+  {set_vec_element_field_t<VEC_ITEM - 1>::set(in_out, c); in_out.VEC_FIELD = c;}  \
+  }
+
+SET_VEC_ELEMENT_SPEC(1, y);
+SET_VEC_ELEMENT_SPEC(2, z);
+SET_VEC_ELEMENT_SPEC(3, w);
+
+template <typename VecType, typename Type, int VecItem>
+struct set_vec_element_t
+{
+  std::size_t n;
+
+  set_vec_element_t(std::size_t n)
+    : n(n)
+  {}
+
+  template <typename CounterT>
+  __device__ VecType operator()(CounterT id)
+  {
+    VecType output;
+    set_vec_element_field_t<VecItem>::set(output, static_cast<Type>(id)%n);
+    return output;
+  }
+};
+
+#define VEC_SET_SPECIALIZATION(TYPE, SCALAR_TYPE, SIZE) \
+template<> void gen<TYPE##SIZE>(modulo_t mod, \
+                                thrust::device_vector<TYPE##SIZE> &data) \
+{ \
+  thrust::tabulate( \
+    thrust::device, \
```

Comment:
You are providing `thrust::device_vector`, do we have to specify execution policy explicitly here?